lmdeploy/pytorch/engine/engine.py

```diff
@@ -828,16 +827,15 @@ def __need_logits(seqs: SeqList):
             return any(seq.return_logits for seq in seqs)
 
         while True:
-            is_prefill = await in_que.get()
+            is_prefill, scheduler_output = await in_que.get()
             try:
+                running = scheduler_output.running
+                adapters = scheduler_output.adapters
+                swap_in_map = scheduler_output.swap_in_map
+                swap_out_map = scheduler_output.swap_out_map
                 prefill_interval = self.scheduler_config.prefill_interval
-                schedule_output = self.scheduler.schedule(
-                    is_prefill=is_prefill, prealloc_size=prefill_interval)
-                running: SeqList = schedule_output.running
-                adapters = schedule_output.adapters
                 loop_count = 1 if is_prefill else (prefill_interval - 1)
-                if len(running) == 0:
-                    raise NoRunningSeqs()
+                assert len(running) > 0
```

Comment:
how do we make sure it is True?

Reply:
https://github.com/InternLM/lmdeploy/blob/9cb25830bf61a86fbf4bf4a4de801ecea4a38d31/lmdeploy/pytorch/engine/engine.py#L934-L936

Empty requests would be skipped.